trace.go

```diff
@@ -138,12 +138,50 @@ func GetFields(err error) map[string]interface{} {
 	if err == nil {
 		return map[string]interface{}{}
 	}
+
+	// for proxyError combine top-level and nested fields.
+	if proxy, ok := err.(proxyError); ok {
+		fields := map[string]interface{}{}
+
+		// nested
+		for field, value := range GetFields(proxy.Err) {
+			fields[field] = value
+		}
+
+		// top-level
+		for field, value := range proxy.GetFields() {
+			fields[field] = value
+		}
+
+		return fields
+	}
+
 	if wrap, ok := err.(Error); ok {
 		return wrap.GetFields()
 	}
+
 	return map[string]interface{}{}
 }
 
+// UnmarshalField attempts to find a given named field.
+// If found, the field is reinterpreted through json.Unmarshal call and stored in unmarshalPtr.
+// Returns true on success and false on failure.
+func UnmarshalField(err error, fieldName string, unmarshalPtr interface{}) bool {
```

Comment:
For me this function looks like an utils function that can  be easily implemented on the client side:
```go
func getSSOErrorDeails(err error) (*SSOErrorDetails, error) {
  raw, ok := trace.GetFeilds(err)[SSODetailsFieldName]
  if !ok {
     return nil, trace.NotFound(...)
  }
  var out SSOErrorDetails
   ...
  return &out, nil
}
```

Reply:
Sure, I could move it to the client-side just fine. I don't have strong feelings about this.